README.md

```diff
@@ -148,13 +152,59 @@ make -j8 install
 | REST Client Example    | `build/Release/bin/rest_client`      |
 
  * A debug library can be created by adding `-DCMAKE_BUILD_TYPE=Debug` to the cmake command above.
-   + The debug library will be put in `build/Debug`.
+   + The debug library will be put in `build/Debug` rather than `build/Release`.
      Use the cmake option `-DCMAKE_INSTALL_PREFIX=../Release` to change this.
 
  * The -j option can be used with the `make install` command to compile with multiple threads.
 
  * This will not build the Python interface. Use the Python build described above to build and install the python interface.
 
+Here is an example of a **Release build** of your own C++ app that links to htm.core as a shared library.
+```
+#! /bin/sh
+# Using GCC on linux ...
+# First build htm.core from sources.
+#      cd <path-to-repo>
+#      mkdir -p build/scripts
+#      cd build/scripts
+#      cmake ../..
+#      make -j4 install
+#
+# Now build myapp
+# We use -std=c++17 to get <filesystem> so we can avoid using the boost library.
+# The -I gives the path to the includes needed to use with the htm.core library.
+# The -L gives the path to the shared htm.core library location at build time.
+# The LD_LIBRARY_PATH envirment variable points to the htm.core library location at runtime.
+g++ -o myapp -std=c++17  -I <path-to-repo>/build/Release/include myapp.cpp -L <path-to-repo>/build/Release/lib -lhtm_core -lpthread -ldl
```

Comment:
this is very nice :+1: do we want to add it as a "demo" step to the CI, or Readme is enough?